flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java

```diff
@@ -147,10 +163,19 @@ public static JobManagerSharedServices fromConfiguration(
                                 failOnJvmMetaspaceOomError ? fatalErrorHandler : null,
                                 checkClassLoaderLeak));
 
+        final int numJobManagerFutureThreads =
+                config.getInteger(
+                        JobManagerOptions.JOB_MANAGER_FUTURE_THREADS, Hardware.getNumberCPUCores());
         final ScheduledExecutorService futureExecutor =
                 Executors.newScheduledThreadPool(
-                        Hardware.getNumberCPUCores(),
-                        new ExecutorThreadFactory("jobmanager-future"));
+                        numJobManagerFutureThreads, new ExecutorThreadFactory("jobmanager-future"));
+
+        final int numJobManagerIoThreads =
+                config.getInteger(
+                        JobManagerOptions.JOB_MANAGER_IO_THREADS, Hardware.getNumberCPUCores());
```

Comment:
nit: `Hardware.getNumberCPUCores` could be deduplicated.

Reply:
deduplicating.